src/openrpc/media_access.json

```diff
@@ -0,0 +1,580 @@
+    "methods": [
+        {
+            "name": "volumes",
+            "tags": [
+                {
+                    "name": "temporal-set"
+                },
+                {
+                    "name": "capabilities",
+                    "x-uses": [
+                        "xrn:firebolt:capability:media-access:volumes"
+                    ]
+                }
+            ],
+            "summary": "Access available Volumes with MediaFiles.",
+            "params": [
+                {
+                    "name": "query",
+                    "required": true,
+                    "schema": {
+                        "$ref": "#/components/schemas/Volume"
+                    }
+                }
+            ],
+            "result": {
+                "name": "volumes",
+                "schema": {
+                    "type": "array",
+                    "items": {
+                        "title": "Volume",
+                        "$ref": "#/components/schemas/Volume"
+                    }
+                }
+            },
+            "examples": [
+                {
+                    "name": "Default example #1",
+                    "params": [
+                        {
+                            "name": "query",
+                            "value": {
+                                "type": "all"
+                            }
+                        }
+                    ],
+                    "result": {
+                        "name": "volumes",
+                        "value": [
+                            {
+                                "uri": "/usb/my-usb-drive",
+                                "name": "My USB Drive",
+                                "type": "usb"
+                            },
+                            {
+                                "uri": "/local/photos",
+                                "name": "photos",
+                                "type": "local"
+                            }
+                        ]
+                    }
+                }
+            ]
+        },
+        {
+            "name": "audio",
+            "tags": [
+                {
+                    "name": "capabilities",
+                    "x-uses": [
+                        "xrn:firebolt:capability:media-access:audio"
+                    ]
+                }
+            ],
+            "summary": "Access available audio MediaFiles.",
+            "params": [
+                {
+                    "name": "volumes",
+                    "summary": "A set of queries used to match which volumes will be matched.",
+                    "required": false,
+                    "schema": {
+                        "$ref": "#/components/schemas/Volume"
+                    }
+                },
+                {
+                    "name": "files",
+                    "summary": "A set of queries used to match which files will be matched.",
+                    "required": false,
```

Comment:
you have two optional params for this method, that is not possible with javascript. You will need to wrap this in an object so that it has a single parameter that and these can be passed as optional fields on that object.

I forgot awhile back I was going to propose openrpc detects this and throws a validation error.

Reply:
it's absolutely possible. It just means you can't leave out the first parameter if you want to use the second one.

I'll take a look at the API and see if we can make the first one required, though.